### ISSUE TITLE
Subir escadas

### ISSUE THREAD

**Issue by Sestuna-bcl** [Expansion-Pack/Harvest-Sky#26]
Precisamos da arte das escadas/cordas/estruturas de subir e uma possível animação do personagem.

**Comment by OttoRobba** [comment]
Sou favorável a fazer o personagem pular mais alto, bem plataforma mesmo, pra não precisar animar descendo/subindo. Assim dá focar nos tiles, crops e UI

**Comment by Sestuna-bcl** [comment]
sim, é uma boa. Mas ai temos o role de como vamos expandir as ilhas.
Porque o esquema é se o personagem atravessa a ilha de cima quando pula por baixo, tipo super mario.  e como ele faz pra descer. Se nao tiver um buraco ou ponto pra descer, ele tera que desce com seta pra baixo + pulo, ou duas setas pra baixo. Se tiver um buraco, precisamos prever de ele nao pode ser preenchido quando o usuário comprar novos tiles.

**Comment by OttoRobba** [comment]
Hmm fato. O que poderíamos fazer é que as ilhas estivessem um pouco pra cima. O jogador deve construir "degraus" com tiles mesmo. Meio que entraria pela esquerda, desceria pela direita manja?

Eu estava pensando no editor de tiles e não seria tão difícil - se a pessoa pode clicar onde quiser, acho que fica "fácil" fazer o esquema acima. Trabalheira mas enfim

O que você acha?

**Comment by OttoRobba** [comment]
Tipo isso:

![plat](https://cloud.githubusercontent.com/assets/597287/6049620/4e8215be-aca0-11e4-8798-9d27b223f2da.png)

**Comment by OttoRobba** [comment]
Daria pra guardar quantos tiles um tile está tocando pra determinar se dá ou não pra construir um tile novo. Não seria um algoritmo muito difícil.

**Comment by OttoRobba** [comment]
Ou até, a pessoa pode crescer apenas os tiles em linha reta pra direita. Daí tem tiles vazios flutuando que se ligam com o da pessoa, daí a escada "já está feita". Seria bem tranquilo fazer isso, só precisaríamos então pré-determinar a posição das ilhas e tal

**Comment by Sestuna-bcl** [comment]
o problema de clicar onde quiser colocar a tile é se o cara estiver no controle e nao com teclado e mouse, pq ai ele tera que percorrer todo o editor usando o direcional. E quem esta rodando o jogo num mobile? como percorre?

Se a ideia é usar o pulo pra que o jogador suba direto pra plataforma, nao tem porque fazer a escada. 
Mas ai temos que pensar que a posição das ilhas tem que ser pensada de um jeito que ele possa pular de uma pra outra.

Ele consegue expandir as ilhas dos NPCS? eu tinha pensado que sim quando citei o lance de andares, pq ai ele nao cria diagonais. Usando a sua imagem como exemplo, os tiles rosa de baixo são o andar 0, o andar que o jogador começa. Já a ilha do npc esta no andar 1, e quando ele for comprar novos tiles escolhe o andar e a direção. O problema desse metodo é o jogador se atrapalhar e acabar criando tiles que bloqueiam seu pulo em algum momento. (tipo o andar de cima ser maior que o de baixo e ele nao ter por onde subir.

Sobre a posição das ilhas, caimos naquele esquema que falamos de objetivos. Se o negocio é tentar linkar as ilhas o mais rapido possivel, entao tem que estar claro pro jogador onde estao as ilhas. Assim ele fica feliz por ter linkado com apenas 50 tiles, por exemplo. 
Agora se o objetivo é só passear por ai e descobrir o mundo, ai podemos ter ilhas em posições aleatorias.

**Comment by OttoRobba** [comment]
Pra percorrer, no caso, a gente teria que guardar a posição do grid selecionado - essa parte não é tão tensa. É similar ao que eu fiz nesse menu: http://invrse.com.br/create-menu-screen-in-phaser/

O que eu quis dizer por escada, seria fazer "degraus" de tile.

Tipo:
    -------
    ------X
    ----XX
    --XXX

Que daí pula estilo plataforma mesmo.

Não sei como cuidar da expansão, honestamente. Acho legal expandir das ilhas que encontra.
O objetivo deve ser, acredito, achar todos os npcs em suas ilhas - daí a pessoa escolhe ir com calma ou ir rápido.

**Comment by Sestuna-bcl** [comment]
hum, com essa mecânica acho que a coisa pode ser bem simples pro jogador.
Também acho que o objetivo do jogo seja encontrar todos os npcs.

**Comment by LidiaYoshino** [comment]
seria muito dificil fazer o personagem passar na frente dos blocos sem colisão se ele só andar, mas ter colisão se ele pular? Tipo isso:

![mechanicaescada](https://cloud.githubusercontent.com/assets/10767442/6115789/4bb7fba8-b08f-11e4-80ac-34aac3a2a323.png)

![mechanicaescada2](https://cloud.githubusercontent.com/assets/10767442/6115791/54ad2210-b08f-11e4-858b-0172d4151fed.png)

**Comment by OttoRobba** [comment]
Fica um teco mais difícil mas não tanto, eu precisaria investigar como fazer essa colisão "nway".

Um problema que surge, creio, é o jogador plantar num tile que está sobreposto. Graficamente acho que ficaria tumultuado.

**Comment by LidiaYoshino** [comment]
a gente pode resolver isso plantando a escada e inutilizando os blocos abaixo dela:

![mechanicaescada3](https://cloud.githubusercontent.com/assets/10767442/6116402/72f38f30-b093-11e4-8165-4ce4c57e58c1.jpg)

**Comment by Sestuna-bcl** [comment]
A ideia de plantar a escada é bem legal. O que pode ser um problema é como fazer cada crop de escada ter um tamanho diferente. Serao sementes diferentes pra cada altura? O mais interessante seria o crop interagir com o anterior e caso seja uma escada, crescer um tile de altura a mais. Outro negocio é explicar pro jogador quantos crops ele tem que plantar pra ter uma escada completa de um andar. Pq ele pode acabar plantando a mais ou a menos, e acho dificil ele interagir com esse crop se vamos travar a tile dele depois de plantado. Se a interacao estiver ok, ele pode colher um degrau errado e replantar.

**Comment by LidiaYoshino** [comment]
Acho q a idéia era plantar o bloco inteiro já. Quando você planta a escada vc ja seleciona 4 tiles e a direção q a escada vai. Podemos fazer também que escadas na beirada da plataforma geram 4 tiles pra escada, mas custam mais caro.
